### PR TITLE
ospath: port ospath.child func from windmill

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @dbentley,

Please review the following commits I made in branch maiamcc/ospath-child:

7226680a2c994269bc392e520daa04816f6b380d (2018-08-16 18:40:06 -0400)
add test

847c13a1348efa7f3005a961011c60bb4b0b7160 (2018-08-16 18:21:55 -0400)
test util for TempDirFixture, redo docker_test with it

10c359255fa762c153e755a5b2223b731ccd37b6 (2018-08-16 18:21:37 -0400)
add ospath.child

### CODE REVIEW

File: internal/build/docker_test.go [hunk 1]
```diff
@@ -19,14 +18,14 @@ import (
 	"time"
 
 	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/test_utils"
```
Comment (nicks) [inline internal/build/docker_test.go]: package names should not have underscores
https://golang.org/doc/effective_go.html#package-names

File: internal/ospath/ospath.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	for true {
+		if dir == current {
+			return child, true
+		}
+
+		if len(current) <= len(dir) || current == "." {
+			return "", false
+		}
+
+		cDir := util.Dir(current)
```
Comment (nicks) [inline internal/ospath/ospath.go]: i would probably use filepath.Dir, filepath.Base, and filepath.Join directly

File: internal/test_utils/temp_dir_fixture.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package test_utils
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/windmilleng/wat/os/temp"
```
Comment (nicks) [inline internal/test_utils/temp_dir_fixture.go]: wat

i think you meant the one in wmclient

Comment (maiamcc) [reply internal/test_utils/temp_dir_fixture.go]: lol good catch

**Pull request merged**